### PR TITLE
Remove 4 minute timeout for log follow process

### PR DESCRIPTION
**Pull request by ckadner**
The process that follows the training logs of an ongoing training job
should not timeout after 4 minutes. Instead the log follow process
should complete after the training job itself is finished.

This behavior is necessary to enable chaining up commands to create
machine learning pipelines, where subsequent commands require the output
data of the training job whose logs are being "followed" like in our
[ART notebook](https://github.com/IBM/FfDL/tree/master/etc/notebooks/art).

This commit reinstates the log follow behavior prior merge of PR #79

See issue #106

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

**Comment by ckadner** [comment]
@animeshsingh
> logs start streaming after 4 minutes

no, that's independent from this timeout

**Comment by ckadner** [comment]
@Tomcli @fplk -- would you know why the `logs --follow` behavior was changed in PR #79 and could you comment on this propose change to reinstate the pre-PR #79 bahavior?

File: trainer/trainer/trainer_impl.go [hunk 1]
```diff
@@ -1403,7 +1403,14 @@ func (s *trainerService) GetTrainedModelLogs(req *grpc_trainer_v2.TrainedModelLo
...
 
-	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*4)
+	var ctx context.Context
+	var cancel context.CancelFunc
+	logr.Debugf("follow is %t", req.Follow)
+	if req.Follow {
+		ctx, cancel = context.WithTimeout(context.Background(), 10*(time.Hour*24))
+	} else {
+		ctx, cancel = context.WithTimeout(context.Background(), 5*time.Second)
+	}
```
Comment (sboagibm) [inline trainer/trainer/trainer_impl.go]: @ckadner can you hold off on reverting this section of code, and try the solution proposed in https://github.com/IBM/FfDL/issues/106#issuecomment-401844947 first?

**Comment by ckadner** [comment]
@sboagibm -- I updated this PR according to your [change request](https://github.com/IBM/FfDL/issues/106#issuecomment-401844947). Thanks for your review!

**Pull request merged**